src/scripts/data_file_processing/healthsystem/equipment/equipment_availability_estimation.py

```diff
@@ -0,0 +1,273 @@
+# -------------------------------------------------------------------------------------------------------------------
+# Final formatting for compatability with the `Equipment` class
+
+final_equipment_availability_export = final_equipment_availability.copy()
+
+# - Drop any rows with missing data
+final_equipment_availability_export = final_equipment_availability_export.dropna()
+
+# - 'Item_Code' is the name expected in the Equipment class, and this should of type `int`
+final_equipment_availability_export['Item_Code'] = final_equipment_availability_export['Item_code'].astype(int)
+
+# Rename 'facility_level'
+final_equipment_availability_export = final_equipment_availability_export.rename(columns={'Facility_level': 'Facility_Level'})
+
+# Check that every row specifies a facility_level
+assert final_equipment_availability_export['Facility_Level'].isin([
+    '0', '1a', '1b', '2', '3', '4', '5'
+]).all()
+assert final_equipment_availability_export['Facility_Level'].notnull().all()
+
+# - 'Pr_Available' is the name expected in the Equipment class, and is taken to be the probability that an item of equipment is available for use
+#   which is taken to be equivalent to it to being "functional"
+final_equipment_availability_export['Pr_Available'] = final_equipment_availability['functional'].astype(float)
+assert final_equipment_availability_export['Pr_Available'].notnull().any()
+
+# - Specify region (needed for merging in facility_ids)
+region_lookup = pd.read_csv(resourcefilepath / 'demography/ResourceFile_PopulationSize_2018Census.csv').set_index('District')['Region'].to_dict()
+final_equipment_availability_export['Region'] = final_equipment_availability_export['District'].map(region_lookup)
+
+# -- Drop not needed columns
+final_equipment_availability_export = final_equipment_availability_export[[
+    'Facility_Level', 'Region', 'District', 'Item_Code', 'Pr_Available'
+]]
+
+# - Merge in 'Facility_ID' (which specifies the level and the district of the facilities)
+mfl = pd.read_csv(
+    resourcefilepath / 'healthsystem' / 'organisation' / 'ResourceFile_Master_Facilities_List.csv'
+)
+
+# Merge in district-specific facilities (matching on level and district)
+final_equipment_availability_export = final_equipment_availability_export.merge(
+    mfl[['Facility_ID', 'Facility_Level', 'District']],
+    left_on=['Facility_Level', 'District'],
+    right_on=['Facility_Level', 'District'],
+    how='left',
+)
+
+# Merge in region-level facilities (level 3) (matching on region only)
+
+# - First, ensure that there is only one set of facility_level 3 entries for each region. (In the current dataframe
+#   entries for level 3 of Blantyre City and Zomba City could both inform on level 3 in the Southern region.)
+#   Do this by taking the mean of Pr_Available across those rows that are facility_level 3 in the same region.
+#   Create a 'reconciled' set of data facility level 3.
```

Comment:
As facility_level = 3 is defined for the region, not the district, we needed to do this reconciliation here. Does it seem ok to you @sakshimohan ?

Reply:
Ah I didn't know this was the case - noted for future analyses. The reconciliation looks good.